src/causalprog/graph/node.py

```diff
@@ -81,3 +93,29 @@ def sample(
 
     def __repr__(self) -> str:
         return f'DistributionNode("{self.label}")'
+
+
+class ParameterNode(Node):
+    """A node containing a parameter."""
+
+    def __init__(
+        self, label: str, *, value: float | None = None, is_outcome: bool = False
+    ) -> None:
+        """Initialise."""
+        super().__init__(label, is_outcome=is_outcome, is_parameter=True)
+        self.value = value
+
+    def sample(
+        self,
+        sampled_dependencies: dict[str, npt.NDArray[float]],  # noqa: ARG002
+        samples: int,
+        rng_key: jax.Array,  # noqa: ARG002
+    ) -> npt.NDArray[float]:
+        """Sample a value from the node."""
+        if self.value is None:
+            msg = "Cannot sample an undetermined parameter node."
+            raise ValueError(msg)
+        return np.full(samples, self.value)
```

Comment:
Do we want to return a `jax.numpy.full` here? I guess right now it's OK as a `numpy` return since that's what our other algorithms are returning too, but we should probably move to using `jax.numpy` everywhere for future proofing.

Reply:
https://github.com/UCL/causalprog/issues/44